dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -90,11 +98,22 @@
     {%- call statement('main', language=language) -%}
       {{ create_table_as(False, target_relation, compiled_code, language) }}
     {%- endcall -%}
+
   {% elif full_refresh_mode %}
+    {% if target_relation.needs_to_drop(existing_relation) %}
+      {{ drop_relation_if_exists(existing_relation) }}
+    {% endif %}
     {%- call statement('main', language=language) -%}
       {{ create_table_as(False, target_relation, compiled_code, language) }}
     {%- endcall -%}
 
+  {% elif target_relation.table_format != existing_relation.table_format %}
```

Comment:
We may ultimately change this behavior (although I'm lobbying for it right now)

Reply:
We're doing this for now! We will be looking to smoothly transition tables between formats with the advent of more catalog formalizations in dbt.